### PR TITLE
cluster-api-provider: create an image

### PR DESCRIPTION
**Pull request by nunnatsa**

### CODE REVIEW

File: ci-operator/config/openshift/cluster-api-provider-kubevirt/openshift-cluster-api-provider-kubevirt-main.yaml [hunk 1]
```diff
@@ -4,6 +4,70 @@ build_root:
...
+    COPY --from=builder /workspace/manager .
+    # Use uid of nonroot user (65532) because kubernetes expects numeric user when applying pod security policies
+    USER 65532
+    ENTRYPOINT ["/manager"]
+  from: base
+  inputs:
+    openshift_release_golang-1.17:
+      as:
+      - registry.ci.openshift.org/openshift/release:golang-1.17
+  to: cluster-api-kubevirt-controller
```
Comment (davidvossel) [inline ci-operator/config/openshift/cluster-api-provider-kubevirt/openshift-cluster-api-provider-kubevirt-main.yaml:66]: for consistency with other openshift forks, i think this should be called `kubevirt-machine-controllers"

File: ci-operator/config/openshift/cluster-api-provider-kubevirt/openshift-cluster-api-provider-kubevirt-main.yaml [hunk 2]
```diff
@@ -1,9 +1,82 @@
...
     name: release
     namespace: openshift
     tag: golang-1.17
+images:
+- build_args:
+  - name: ARCH
+    value: amd64
+  - name: ldflags
+  dockerfile_literal: |-
+    FROM registry.ci.openshift.org/openshift/release:golang-1.17 as toolchain
```
Comment (davidvossel) [inline ci-operator/config/openshift/cluster-api-provider-kubevirt/openshift-cluster-api-provider-kubevirt-main.yaml:22]: I wonder if we should be following the pattern that other providers are using here?

example - https://github.com/openshift/cluster-api-provider-aws/blob/release-4.10/Dockerfile.rhel

File: ci-operator/config/openshift/cluster-api-provider-kubevirt/openshift-cluster-api-provider-kubevirt-main.yaml (hunk 1 above)
Comment (nunnatsa) [reply ci-operator/config/openshift/cluster-api-provider-kubevirt/openshift-cluster-api-provider-kubevirt-main.yaml:66]: I copied it from AWS repo

Comment (nirarg) [reply ci-operator/config/openshift/cluster-api-provider-kubevirt/openshift-cluster-api-provider-kubevirt-main.yaml:66]: @davidvossel the `aws-machine-controller` term is used for the code and images used for `openshift/machine-api-operator`
capk code is not used for `openshift/machine-operator` therefore we shouldn't use the same term
Also this image shouldn't be added to the release image
Therefore I agree that currently we should build the image the same way it done in `capa` for hypershift
This is the relevant PR: https://github.com/openshift/release/pull/23913
Please notice that in `openshift/cluster-api-provider-aws` there are two different images:
1. The image used for `openshift/machine-api-operator` based on the code from the master
2. The image for hypershift capa - based on code from upstream, special branches are created from upstream

Comment (davidvossel) [reply ci-operator/config/openshift/cluster-api-provider-kubevirt/openshift-cluster-api-provider-kubevirt-main.yaml:55]: got it, makes sense. I see what you're saying is consistent with the aws controllers consumed by hypershift. thanks for the explanation

```
        // Image built from https://github.com/openshift/cluster-api-provider-aws/tree/release-1.1
        // Upstream canonical image comes from  https://console.cloud.google.com/gcr/images/k8s-artifacts-prod
        // us.gcr.io/k8s-artifacts-prod/cluster-api-aws/cluster-api-aws-controller:v1.1.0
        imageCAPA = "registry.ci.openshift.org/hypershift/cluster-api-aws-controller:v1.1.0"

```

**Review by davidvossel — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**